homeassistant/components/flume/sensor.py

```diff
@@ -40,25 +47,59 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     flume_token_file = hass.config.path("FLUME_TOKEN_FILE")
     time_zone = str(hass.config.time_zone)
     name = config[CONF_NAME]
+    include_devices = [
+        str(device_id) for device_id in config.get(CONF_INCLUDE_DEVICES, [])
+    ]
+    exclude_devices = [
+        str(device_id) for device_id in config.get(CONF_EXCLUDE_DEVICES, [])
+    ]
     flume_entity_list = []
 
+    http_session = Session()
+
     flume_devices = FlumeDeviceList(
-        username, password, client_id, client_secret, flume_token_file
+        username,
+        password,
+        client_id,
+        client_secret,
+        flume_token_file,
+        http_session=http_session,
     )
 
     for device in flume_devices.device_list:
         if device["type"] == FLUME_TYPE_SENSOR:
+            device_id = device["id"]
+            device_name = device["location"]["name"]
+            if exclude_devices and str(device_id) in exclude_devices:
+                _LOGGER.info(
+                    "Skipping device %s (%s) because it is in exclude_devices",
+                    device_id,
+                    device_name,
+                )
+                continue
+            if include_devices and str(device_id) not in include_devices:
+                _LOGGER.info(
+                    "Skipping device %s (%s) because it is not in include_devices",
+                    device_id,
+                    device_name,
+                )
+                continue
+
             flume = FlumeData(
```

Comment:
I know that this is not part of this PR, But a data object per sensor? That feels like code smell. How do the data objects share an updated token in `flume_token_file` ? 

Looks like the lib could use a refactor as described here https://developers.home-assistant.io/docs/api_lib_auth

Reply:
(btw, I notice that it's OAuth2. We should consider adding the password grant to our OAuth2 helper)

Reply:
@balloob. Is Somfy the best example of this implementation?

Reply:
Will see about fixing the object per sensor.  I’m not sure how available upstream (@ChrisMandich) will be to review PRs as I haven’t heard back yet

Reply:
@bdraco appreciate the feedback and like the direction you are going. Reviewed the PR you provided, I see no issue with approving it. Seems like there are a few asks from balloob which can bee future additions with a future change?

Reply:
@ChrisMandich Thanks for reviewing!  Once you get 0.3.0 published I'll reduce this down based on @balloob 's feedback and work on another PR to additional items.